daemon/policy.go

```diff
@@ -244,6 +244,14 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 		}
 	}
 	rev := d.policy.AddListLocked(rules)
+
+	// The rules are added, we can begin ToFQDN DNS polling for them
+	// Note: api.FQDNSelector.sanitize checks that the matchName entries are
+	// valid. This error should never happen (of course).
+	if err := d.dnsRuleGen.StartManageDNSName(rules); err != nil {
```

Comment:
Why is the move needed? I see:
```
func (gen *RuleGen) StartManageDNSName(sourceRules []*api.Rule) error {
        gen.Lock()
        defer gen.Unlock()
```

Is it that `StartManageDNSName()` must see the rules in the policy repository?